tests/integration/UploadWordLinkRecordingIntegrationTest.php

```diff
@@ -0,0 +1,228 @@
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT_UPDATED),
+        ];
+
+
+        // update should not create a new recording
+        $this->assertEquals($recordingId, $this->sendRequestAndReturnRecordingId($updateRecordingPayload));
+
+        // assert old recording file is deleted
+        $createdRecordingFile = sprintf("%s/%s/WordLinks/%s", self::$uploadedProjectDir, $createRecordingPayload['PhoneId'], $createdRecording['fileName']);
+        $this->assertFileDoesNotExist($createdRecordingFile);
+
+
+        # assert updated recording data
+        $updateRecording = $this->getRecording($recordingId);
+        $this->assertEquals(self::TEXT_BACK_TRANSLATION_CONTENT_UPDATED, $updateRecording['textBackTranslation']);
+        
+        // assert recording file name is updated
+        $this->assertNotEquals($updateRecording['fileName'], $createdRecording['fileName']);
+
+        // verify new recording file is uploaded successfully
+        $updatedRecordingFile = sprintf("%s/%s/WordLinks/%s", self::$uploadedProjectDir, $updateRecordingPayload['PhoneId'], $updateRecording['fileName']);
+        $this->assertEquals(self::RECORDING_FILE_EXTENSION, pathinfo($updateRecording['fileName'], PATHINFO_EXTENSION));
+        $this->assertFileExists($updatedRecordingFile);
+        $this->assertEquals(self::AUDIO_FILE_CONTENT_UPDATED, file_get_contents($updatedRecordingFile));
+    }
+
+    /**
+     * @test
+     * @throws GuzzleException
+     * @throws \Exception
+     */
+    public function testShouldNotUpdateGivenRecordingIfTermIsChanged() {
+        $createRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_3,
+            'term' => self::TERM_1,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' =>self::RECORDING_FILE_NAME_1,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT),
+        ];
+
+        $recordingId = $this->sendRequestAndReturnRecordingId($createRecordingPayload);
+        $createdRecording = $this->getRecording($recordingId);
+
+        // request payload
+        $updateRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_3,
+            'term' => self::TERM_2,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' => self::RECORDING_FILE_NAME_2,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT_UPDATED
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT_UPDATED),
+            'RecordingId' => $recordingId
+        ];
+
+        // should create a new recording
+        $updatedRecordingId = $this->sendRequestAndReturnRecordingId($updateRecordingPayload);
+        $this->assertNotEquals($createdRecording['id'], $updatedRecordingId);
+
+        # verify exiting recording data isn't changed
```

Comment:
Here, you're verifying the $createdRecording that was loaded before sending $updateRecordingPayload, so nothing would change. You can remove this section. 

Instead, please verify the WordlinkRecordings has been updated successfully.